template/registry_datetime.go

```diff
@@ -183,6 +184,35 @@ func parseDatetimeOptions(options Options) (*datetimeOptions, error) {
 	return &opts, nil
 }
 
+// A conflictingOptions checks if options conform to WG registry spec:
+//
+//	https://github.com/unicode-org/message-format-wg/blob/main/spec/registry.md#options-3
+//
+// The :datetime function can use either the appropriate style options or can use a collection
+// of field options (but not both) to control the formatted output.
+// If both are specified, a Bad Option error MUST be emitted and a fallback value used as the
+// resolved value of the expression.
+func conflictingOptions(options datetimeOptions) error {
+	hasStyleOptions := options.TimeStyle != "" || options.DateStyle != ""
+	hasFieldOptions := options.Weekday != "" ||
+		options.Era != "" ||
+		options.Year != "" ||
+		options.Month != "" ||
+		options.Day != "" ||
+		options.Hour != "" ||
+		options.Minute != "" ||
+		options.Second != "" ||
+		options.FractionalSecondDigits != 0 ||
+		options.HourCycle != "" ||
+		options.TimeZoneName != ""
+
+	if hasStyleOptions && hasFieldOptions {
+		return errors.New("bad option")
+	}
```

Comment:
We don't need to compute both - we can return sooner and speed things up.

```suggestion
	hasStyleOptions := options.TimeStyle != "" || options.DateStyle != ""
	if !hasNoStyleOptions {
	   return nil
	}
	
	hasFieldOptions := options.Weekday != "" ||
		options.Era != "" ||
		options.Year != "" ||
		options.Month != "" ||
		options.Day != "" ||
		options.Hour != "" ||
		options.Minute != "" ||
		options.Second != "" ||
		options.FractionalSecondDigits != 0 ||
		options.HourCycle != "" ||
		options.TimeZoneName != ""
	if hasFieldOptions {
		return errors.New("bad option")
	}
```